Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide/Internal.hs

```diff
@@ -119,6 +149,32 @@ endRideTransaction driverId bookingId ride mbFareParams mbRiderDetailsId merchan
     updateDriverWeeklyZscore ride rideDate weekStartDate weekEndDate driverWeeklyZscore ride.chargeableDistance
   DLoc.updateOnRideCacheForCancelledOrEndRide driverId merchantId
   SRide.clearCache $ cast driverId
+  unless (driverRideFee <= 0) $ do
+    maxShards <- asks (.maxShards)
+    isPendingPaymentJobScheduled <- getPendingPaymentCache driverFee.endTime
+    let pendingPaymentJobTs = diffUTCTime driverFee.endTime now
```

Comment:
why not use `driverPaymentCycleDuration` config here ? this looks weird.

Reply:
Cycle is different, this is the time after which the job will be triggered. I want to trigger it at endTime of window